src/mon/Monitor.h

```diff
@@ -348,7 +348,21 @@ class Monitor : public Dispatcher,
                    start(ceph_clock_now()) { }
     virtual ~ScrubState() { }
   };
-  std::shared_ptr<ScrubState> scrub_state; ///< keeps track of current scrub
+
+  struct ScrubContext {
+    ScrubState scrub_state;       ///< keeps track of current scrub
+    version_t scrub_version;      ///< paxos version we are scrubbing
+    std::map<int,ScrubResult> scrub_result;  ///< result so far
+    ScrubContext() {
+      scrub_version = 0;
+      scrub_result.clear();
+    }
+    ~ScrubContext() {
+      scrub_version = 0;
+      scrub_result.clear();
+     }
+  };
+  boost::atomic_shared_ptr<ScrubContext> scrub_con; ///< keeps track of scrub_context
```

Comment:
nit: is there any reason for not naming it as `scrub_ctx`? `_con` suggest rather a _connection_ which is misleading.

Reply:
done